src/main/java/frc/trigon/robot/subsystems/turret/simulationturret/SimulationTurretIO.java

```diff
@@ -0,0 +1,57 @@
+package frc.trigon.robot.subsystems.turret.simulationturret;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import frc.trigon.robot.constants.RobotConstants;
+import frc.trigon.robot.subsystems.turret.TurretIO;
+import frc.trigon.robot.subsystems.turret.TurretInputsAutoLogged;
+
+public class SimulationTurretIO extends TurretIO {
+    private final DCMotorSim motor = SimulationTurretConstants.MOTOR;
+
+    private double motorVoltage;
+
+    @Override
+    protected void updateInputs(TurretInputsAutoLogged inputs) {
+        motor.update(RobotConstants.PERIODIC_TIME_SECONDS);
+
+        inputs.motorVoltage = motorVoltage;
+        inputs.motorAngleDegrees = getMotorAngleDegrees();
+        inputs.motorVelocityDegreesPerSecond = getMotorVelocityDegreesPerSecond();
+    }
+
+    @Override
+    protected void setTargetAngle(Rotation2d targetAngle) {
+        setMotorVoltageFromPower(calculateMotorPower(targetAngle));
+    }
+
+    @Override
+    protected void stop() {
+        setMotorVoltageFromPower(0);
+    }
+
+    private double calculateMotorPower(Rotation2d targetAngle) {
+        double pidOutput = SimulationTurretConstants.PROFILED_PID_CONTROLLER.calculate(targetAngle.getDegrees());
+        double feedforward = SimulationTurretConstants.FEEDFORWARD.calculate(motor.getAngularPositionRad(), motor.getAngularVelocityRadPerSec(), SimulationTurretConstants.PROFILED_PID_CONTROLLER.getGoal().velocity);
```

Comment:
Since the Phoenix 6 units are revolutions, and we want to match the closest gains to the real gain, make sure you're comparing revolutions and not degrees/rads.